app/controllers/katello/api/v2/sync_plans_controller.rb

```diff
@@ -87,18 +87,12 @@ def destroy
     param_group :search, Api::V2::ApiController
     param :name, String, :desc => N_("product name to filter by")
     def available_products
-      enabled_product_ids = Product.where(:organization_id => @organization).readable.select { |p| p.enabled? }.collect(&:id)
+      product_ids = @sync_plan.product_ids
+      products = Product.enabled.readable
+      products = products.where("#{Product.table_name}.id NOT IN (?)", product_ids) unless product_ids.empty?
 
-      filters = [:terms => {:id => enabled_product_ids - @sync_plan.product_ids}]
-      filters << {:term => {:name => params[:name]}} if params[:name]
-
-      options = {
-        :filters       => filters,
-        :load_records? => true
-      }
-
-      products = item_search(Product, params, options)
-      respond_for_index(:collection => products)
+      options = {:includes => [:sync_plan, :provider], :resource_class => Product}
+      respond_for_index(:collection => scoped_search(products, :name, :desc, options), :template => '../products/index')
```

Comment:
Another option:

c) Do both option (b) and option (a) until the route is deprecated and then remove option (a)

I'll leave it up to you. I have a slight preference for b or c but would be fine with a.